apps/studio/src/features/editing-experience/components/CreateFolderModal/CreateFolderModal.tsx

```diff
@@ -0,0 +1,173 @@
+    mutate({ ...data, parentFolderId, siteId })
+  })
+
+  const [folderTitle, permalink] = watch(["folderTitle", "permalink"])
+
+  useEffect(() => {
+    const permalinkFieldState = getFieldState("permalink")
+    // This allows the syncing to happen only when the page title is not dirty
+    // Dirty means user has changed the value AND the value is not the same as the default value of "".
+    // Once the value has been cleared, dirty state will reset.
+    if (!permalinkFieldState.isDirty) {
+      setValue("permalink", generateResourceUrl(folderTitle), {
+        shouldValidate: !!folderTitle,
+      })
+    }
+  }, [getFieldState, setValue, folderTitle])
+
+  return (
+    <Modal isOpen={isOpen} onClose={onClose}>
+      <ModalOverlay />
+      <ModalContent>
+        <form onSubmit={onSubmit}>
+          <ModalHeader>Create a new folder </ModalHeader>
+          <ModalCloseButton size="sm" />
+          <ModalBody>
+            <VStack alignItems="flex-start" spacing="1.5rem">
+              <FormControl isInvalid={!!errors.folderTitle}>
+                <FormLabel color="base.content.strong">
+                  Folder name
+                  <FormHelperText color="base.content.default">
+                    This will be the title of the index page of your folder.
+                  </FormHelperText>
+                </FormLabel>
+
+                <Input
+                  placeholder="This is a title for your new folder"
+                  {...register("folderTitle")}
+                />
+                {errors.folderTitle?.message ? (
+                  <FormErrorMessage>
+                    {errors.folderTitle.message}
+                  </FormErrorMessage>
+                ) : (
+                  <FormHelperText mt="0.5rem" color="base.content.medium">
+                    {MAX_FOLDER_TITLE_LENGTH - folderTitle.length} characters
+                    left
+                  </FormHelperText>
+                )}
+              </FormControl>
+              <FormControl isInvalid={!!errors.permalink}>
+                <FormLabel color="base.content.strong">
+                  Folder URL
+                  <FormHelperText color="base.content.default">
+                    This will be applied to every child under this folder.
+                  </FormHelperText>
+                </FormLabel>
+                <Input
+                  placeholder="This is a url for your new page"
+                  {...register("permalink")}
+                />
+                {errors.permalink?.message && (
+                  <FormErrorMessage>
+                    {errors.permalink.message}
+                  </FormErrorMessage>
+                )}
+
+                <Box
+                  mt="0.5rem"
+                  py="0.5rem"
+                  px="0.75rem"
+                  bg="interaction.support.disabled"
+                >
+                  <Icon mr="0.5rem" as={BiLink} />
+                  {permalink}
```

Comment:
@sehyunidaaa  - i cheated abit and didn't display the full permalink. i can figure this out later but for now, it's invoked only from 1 site so the full permalink is actually 

`${site.baseUrl}/${permalink}` but we don't store the base url anywhere so...

Reply:
i think this is okay! omitting the base url is fine. if this new folder has a parent, does it show the parent folder's permalink?

Reply:
not for now because we can't access any other folders. i'll fix that + this together